src/core/core.controller.js

```diff
@@ -933,7 +973,26 @@ module.exports = function(Chart) {
 			me.lastActive = me.active;
 
 			return changed;
-		}
+		},
+		/**
+		 * Format the data list if the data is missing some points in the datasets.
+		 * @private
+		 * @param {array} dataArray array to format
+		 * @return {array} the formated array
+		 */
+		_formatDataset: function(dataArray) {
+			var labels = this.chart.data.labels;
+			var tmp = dataArray.slice(0);
+			var result = dataArray;
```

Comment:
this is where you should make the defensive copy. don't modify the input to the function. instead do `var result = dataArray.slice(0); // create copy of input`